test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java

```diff
@@ -684,4 +684,122 @@ public void testResetStreamLog() {
         assertThat(log.getGlobalTail()).isEqualTo(globalTailAfterReset);
         assertThat(log.getTrimMark()).isEqualTo(trimMarkAfterReset);
     }
+
+    @Test
+    public void partialHeaderMetadataTest() throws Exception {
+        String logDir = getContext().getServerConfig().get("--log-path") + File.separator + "log";
+        String logFilePath = logDir + File.separator + 0 + ".log";
+
+        File dir = new File(logDir);
+        dir.mkdir();
+        RandomAccessFile logFile = new RandomAccessFile(logFilePath, "rw");
+
+        Types.LogHeader header = Types.LogHeader.newBuilder()
+                .setVersion(StreamLogFiles.VERSION)
+                .setVerifyChecksum(false)
+                .build();
+
+        // Simulate a partial metadata write for the log header
+        ByteBuffer buf = StreamLogFiles.getByteBufferWithMetaData(header);
+        buf.limit(StreamLogFiles.METADATA_SIZE - 1);
+        logFile.getChannel().write(buf);
+        logFile.close();
+
+        // Open a StreamLog and write an entry in the segment that has the partial metadata write
+        StreamLog log = new StreamLogFiles(getContext(), false);
+        long address0 = 0;
+        assertThat(log.read(address0)).isNull();
+        log.close();
+    }
+
+    @Test
+    public void partialHeaderTest() throws Exception {
+        String logDir = getContext().getServerConfig().get("--log-path") + File.separator + "log";
+        String logFilePath = logDir + File.separator + 0 + ".log";
+
+        File dir = new File(logDir);
+        dir.mkdir();
+        RandomAccessFile logFile = new RandomAccessFile(logFilePath, "rw");
+
+        Types.LogHeader header = Types.LogHeader.newBuilder()
+                .setVersion(StreamLogFiles.VERSION)
+                .setVerifyChecksum(false)
+                .build();
+
+        // Simulate a partial log header write
+        ByteBuffer buf = StreamLogFiles.getByteBufferWithMetaData(header);
+        buf.limit(buf.capacity() - 1);
+        logFile.getChannel().write(buf);
+        logFile.close();
+
+        StreamLog log = new StreamLogFiles(getContext(), false);
+        ByteBuf b = Unpooled.buffer();
+        byte[] streamEntry = "Payload".getBytes();
+        Serializers.CORFU.serialize(streamEntry, b);
+        // Write to segment 0
+        long address0 = 0;
+        log.append(address0, new LogData(DataType.DATA, b));
+        log.close();
+
+        // Open the segment again and verify that the entry write can be read (i.e. log file can be
+        // parsed correctly).
+        log = new StreamLogFiles(getContext(), false);
+        assertThat(log.read(address0).getPayload(null)).isEqualTo(streamEntry);
+    }
+
+    @Test
+    public void partialEntryTest() throws Exception {
```

Comment:
comment.

Reply:
I think you need to add a test with more than one entry. This is a test for one entry only. I want to make sure that we are not off by 1 byte in the calculations.

Reply:
This test does just that, it writes a partial entry, then writes a correct entry and makes sure that it addresses space can be parsed by creating a new stream log object.

I can add more log entries if you think they are necessary.

Reply:
I need  write
1. Entry 1
2. Partial Entry
3. Entry 2
Restart
Read Entry 1 and verify
Read Entry 2 and verify.
If this works we are good the position logic must be correct.

Reply:
Write the test if you want to. Now having looked at the java api docs I think the logic seems ok.

Reply:
Hi @medhavidhawan , I'm not sure how to write such test unless we restart the stream log (i.e. open/close) between writes, which those tests already do. Otherwise you need the channel to throw an IOException during the write which we currently don't have the ability to do with unit tests. For example, I'm not sure how to write a test where the first write succeeds, then the second write will throw an IOException and finally the third write successfully completing. Any ideas?

Reply:
Since IO is controlled by OS the way we can try to achieve this:
1. Mocks: http://dontpanic.42.nl/2012/10/gotta-catch-em-all-100-test-coverage.html
Do not think if this is sufficient for what we want to test. Ok let it be for now. Will try and think of something.